zulip/zulip/__init__.py

```diff
@@ -525,7 +525,10 @@ def error_retry(error_string):
                     sys.stdout.write(".")
                 sys.stdout.flush()
             query_state["request"]["dont_block"] = json.dumps(True)
-            time.sleep(1)
+            delay_cap = 10
+            delay_base = 0.5
+            delay_time = random.random() * min(delay_cap, delay_base * (2 ** query_state["failures"]))
```

Comment:
@orientor Thanks for working on this!  The code looks nice and clean, but I have one concern.

The expression `delay_base * (2 ** query_state{'failures'])` will start overwhelming `delay_cap` by the fifth failure, and then we'll be computing exponential factors of 2 for no reason, and they can get big, well, exponentially.

I think the better strategy here is to tweak it so that after N failures, we just using the `delay_cap` value.  I also think there is some point after which we should simply quit, but that can be for a future PR.

Reply:
See https://chat.zulip.org/#narrow/stream/92-learning/topic/exponential.20backoff.20--.20with.20jitter/near/820056 for more discussion.

Reply:
@showell changed the PR. Now it doesn't compute after 5th power of 2. Regarding stopping do_api_query stops after 10 failures.

Reply:
@showell should I delay_cap and delay_base be added as function parameters?